src/entities/Proposal/jobs.ts

```diff
@@ -107,18 +107,25 @@ async function getProposalsWithOutcome(proposals: ProposalAttributes[], context:
   return pendingProposalsWithOutcome
 }
 
-export async function getFinishableTenderProposals(pendingProposals: ProposalAttributes[]) {
-  let pendingTenderProposals = pendingProposals.filter((item) => item.type === ProposalType.Tender)
-  if (pendingTenderProposals.length > 0) {
-    const linkedProposalIds = [...new Set(pendingTenderProposals.map((item) => item.configuration.linked_proposal_id))]
-    pendingTenderProposals = []
+export async function getFinishabledLinkedProposals(
+  pendingProposals: ProposalAttributes[],
+  type: ProposalType.Bid | ProposalType.Tender
+) {
+  let proposals = pendingProposals.filter((item) => item.type === type)
+  if (proposals.length > 0) {
+    const linkedProposalIds = [...new Set(proposals.map((item) => item.configuration.linked_proposal_id))]
```

Comment:
Why a Set is needed here? Is it possible to have a single "linked_proposal_id" more than once?

Reply:
Yes. This is to get all the linked proposal ids so we can query the db for those proposals.
We could just grab the proposals from the "finishable proposals query" but this was added just to make sure we are getting them all.